task4/src/components/movie/movie.js

```diff
@@ -1,21 +1,44 @@
 import React from 'react';
 import './movie.scss';
 import PropTypes from 'prop-types';
+import EditMovie from '../editMovie/editMovie';
 
-function Movie(props) {
-  const {poster = '', title = '', genre = '', year = ''} = props.movie;
-  return (
-    <div className="movie">
-      <img src={poster}/>
-      <div className="movie-info">
-        <h3>
-          {title}
-          <span>{genre}</span>
-        </h3>
-        <span className="year">{year}</span>
+class Movie extends React.Component{
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      movie: props.movie,
```

Comment:
You do not need to set your movie in the state, you already have it in the prop and you do not change something in your movie object, so just use it in the prop.